### DISCUSSION TITLE
Mason popup doesn't look right

### DISCUSSION THREAD

**Discussion by yigitefe** [mason-org/mason.nvim#865] [Q&A] [answered]
Hey, I am following [this](https://youtu.be/vdn_pKJUda8) tutorial. I am just using a different theme. When I activate `:Mason` I see the popup doesn't look right and bleeds through the status bar. Also when I use `<C-f>` for filtering, it doesn't work and I get a screen like `less` with the list of languages. I am not very experienced with Lua or nvim configurations, can anyone help me fix this? Thanks.

![Screenshot 2023-01-08 at 00 02 24](https://user-images.githubusercontent.com/15162675/211170486-f5f71713-cb69-452f-a6c0-51c6c8adff34.png)
![Screenshot 2023-01-08 at 00 02 33](https://user-images.githubusercontent.com/15162675/211170489-2c84607c-d7f6-46c9-bd94-61f5508a7af2.png)

**Comment by williamboman** [comment, marked as answer]
> Also when I use <C-f> for filtering, it doesn't work and I get a screen like less with the list of languages

Mason uses `vim.ui.select()` to prompt the user for a selection. By default it’ll use the blocking approach that you describe. I personally use [dressing.nvim](https://github.com/stevearc/dressing.nvim) to customize `vim.ui.select`.

**Reply by yigitefe** [reply]
Thank you. Installing dressing.nvim indeed solved that problem. But the popup not showing properly is still an issue.

**Reply by williamboman** [reply]
> But the popup not showing properly is still an issue.

It's... it's a feature 😅. [This](https://github.com/williamboman/mason.nvim/blob/4d734aee41efce4f1a4be7619ad040b53237042e/lua/mason-core/ui/display.lua#L173-L175) is how height is determined. It looks like you have `set cmdheight=0`?

**Reply by williamboman** [reply]
https://github.com/williamboman/mason.nvim/pull/902 should make things look better

**Reply by yigitefe** [reply]
I pulled the update but nothing changed 😄 I set cmdheight to 4 just to compare it, and this is the result. 
![Screenshot 2023-01-14 at 00 43 55](https://user-images.githubusercontent.com/15162675/212425083-32518333-9d63-48ea-ab97-4ef29985b591.png)

P.S. My cmdheight was 1

**Reply by williamboman** [reply]
Hm are you sure? Which commit SHA are you on?

**Reply by yigitefe** [reply]
Yes, I'm sure. This is from `~/.local/share/nvim/site/pack/packer/start/mason.nvim`
![Screenshot 2023-01-14 at 14 37 05](https://user-images.githubusercontent.com/15162675/212470005-ebfb9eed-3ddb-41b9-9c41-afff6306aacb.png)

**Reply by williamboman** [reply]
Ah. It's sort of working as intended, but I see how it looks a bit weird when it overlaps the statusline. https://github.com/williamboman/mason.nvim/pull/905 should (hopefully) improve this

**Reply by yigitefe** [reply]
It definitely improved. Thank you very much. I have a question though, in the screenshots in README, the bottom line of the popup doesn't go all the way to the statusline and it seems to have a different shade than the editor background. What makes the difference? Are there some config params that I can use to tweak the UI?
![Screenshot 2023-01-14 at 15 05 02](https://user-images.githubusercontent.com/15162675/212471106-d4655f87-7ef4-44aa-9cf0-3b24943bb45a.png)

**Reply by williamboman** [reply]
Those screenshots are quite old, the window used to be shorter. As for highlights, you can change it either via the `NormalFloat` highlight group, or `MasonNormal` for the Mason window only, for example:

```vim
:hi MasonNormal guibg=red
```
